hoomd/MeshGroupData.cc

```diff
@@ -0,0 +1,803 @@
+                std::pair<unsigned int, unsigned int> rank_idx = rank_rtag_it->second;
+                unsigned int rank = rank_idx.first;
+                unsigned int idx = rank_idx.second;
+
+                snapshot.type_id[snap_id] = typevals_proc[rank][idx].type;
+		for(unsigned int i = 0; i< group_size_half; i++)
+		    {
+                    snapshot.groups[snap_id].tag[i] = members_proc[rank][idx].tag[i];
+		    }
+                snap_id++;
+                }
+            }
+        }
+    else
+#endif
+        {
+        // allocate memory in snapshot
+        snapshot.resize(this->getNGlobal());
+
+        assert(this->getN() == this->getNGlobal());
+        std::map<unsigned int, unsigned int>::iterator rtag_it;
+        // index in snapshot
+        unsigned int snap_id = 0;
+
+        // loop through active tags
+        std::set<unsigned int>::iterator active_tag_it;
+        for (active_tag_it = this->m_tag_set.begin(); active_tag_it != this->m_tag_set.end(); ++active_tag_it)
+            {
+            unsigned int group_tag = *active_tag_it;
+            rtag_it = rtag_map.find(group_tag);
+            if (rtag_it == rtag_map.end())
+                {
+                this->m_exec_conf->msg->error() << endl
+                                          << "Could not find " << name << " " << group_tag
+                                          << ". Possible internal error?" << endl
+                                          << endl;
+                throw std::runtime_error("Error gathering " + std::string(name) + "s");
+                }
+
+            // store tag in index
+            index.insert(std::make_pair(group_tag, snap_id));
+
+            unsigned int group_idx = rtag_it->second;
+	    members_t member = this->m_groups[group_idx];
+	    for(unsigned int i = 0; i< group_size_half; i++)
+	        {
+                snapshot.groups[snap_id].tag[i] = member.tag[i];
+		}
+            snapshot.type_id[snap_id] = ((typeval_t)this->m_group_typeval[group_idx]).type;
+            snap_id++;
+            }
+        }
+
+    snapshot.type_mapping = this->m_type_mapping;
+
+    return index;
+    }
+
+
+namespace detail
+    {
+template<class T, typename Group>
+void export_MeshGroupData(pybind11::module& m,
+                            std::string name,
+                            std::string snapshot_name,
+                            bool export_struct)
+    {
+    // export group structure
+    if (export_struct)
+        Group::export_to_python(m);
+
+    pybind11::class_<T, std::shared_ptr<T>>(m, name.c_str())
+        .def(pybind11::init<std::shared_ptr<ParticleData>, unsigned int>())
+        .def(pybind11::init<std::shared_ptr<ParticleData>, const typename TriangleData::Snapshot&>())
+        .def("initializeFromSnapshot", &T::initializeFromSnapshot)
+        .def("takeSnapshot", &T::takeSnapshot)
+        .def("addBondedGroup", &T::addBondedGroup);
+    }
```

Comment:
This seems a subset of `export_BondedGroupData`. Could you just use that?

Reply:
It's not a subset. The initialization with `TriangleData::Snapshot` is not used in `export_BondedGroupData`.